pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -3186,3 +3193,62 @@ func getQueuedKeys(queue workqueue.RateLimitingInterface) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+// Controller should not create pods on nodes which have daemon pods, and should remove excess pods from nodes that have extra pods.
+func TestSurgeDealsWithExistingPods(t *testing.T) {
+	ds := newDaemonSet("foo")
+	ds.Annotations = map[string]string{"alpha.apps.openshift.io/surge": "1"}
```

Comment:
Maybe use something other than openshift for in tree k8s. Just a suggestion.

Reply:
heh

Reply:
The downsides of prototyping in a different fork